core/src/language/swift.rs

```diff
@@ -803,20 +792,71 @@ impl Swift {
-        if !decs.contains(&CODABLE.to_string()) {
-            decs.push(CODABLE.to_string());
+        if !decs.contains(&CODABLE) {
+            decs.push(CODABLE);
         }
 
         writeln!(w, "public struct CodableVoid: {} {{}}", decs.join(", "))
     }
+
+    /// Build the generic constraints output. This checks for the `swiftGenericConstraints` typeshare attribute and combines
+    /// it with the `default_generic_constraints` configuration. If no `swiftGenericConstraints` is defined then we just use
+    /// `default_generic_constraints`.
+    fn generic_constraints<'a>(
+        &'a self,
+        decorator_map: &'a DecoratorMap,
+        generic_types: &'a [String],
+    ) -> String {
+        let swift_generic_contraints_annotated = decorator_map
+            .get(&DecoratorKind::SwiftGenericConstraints)
+            .map(|generic_constraints| {
+                generic_constraints
+                    .iter()
+                    .filter_map(|generic_constraint| {
+                        let mut gen_name_val_iter = generic_constraint.split(':');
+                        let generic_name = gen_name_val_iter.next()?;
+                        let mut generic_name_constraints = gen_name_val_iter
+                            .next()?
+                            .split('&')
+                            .map(|s| s.trim())
+                            .collect::<BTreeSet<_>>();
+                        // Merge default generic constraints with annotated constraints.
+                        generic_name_constraints
+                            .extend(self.default_generic_constraints.get_constraints());
+                        Some((generic_name, generic_name_constraints))
+                    })
+                    .collect::<HashMap<_, _>>()
+            })
+            .unwrap_or_default();
+
+        generic_types
+            .iter()
+            .map(
+                |type_name| match swift_generic_contraints_annotated.get(type_name.as_str()) {
+                    // Use constraints from swiftGenericConstraints decorator.
+                    Some(constraints) => (type_name, Either::Left(constraints.iter().copied())),
+                    // Use the default generic constraints if it is not part of a swiftGenericConstraints decorator.
+                    None => (
+                        type_name,
+                        Either::Right(self.default_generic_constraints.get_constraints()),
+                    ),
+                },
+            )
+            .map(|(type_name, mut constraints)| format!("{type_name}: {}", constraints.join(" & ")))
+            .join(", ")
+    }
 }
 
-fn swift_keyword_aware_rename(name: &str) -> String {
-    if SWIFT_KEYWORDS.contains(&name) {
-        return format!("`{}`", name);
+fn swift_keyword_aware_rename(name: Cow<str>) -> Cow<str> {
+    if SWIFT_KEYWORDS.contains(&name.as_ref()) {
+        Cow::Owned(format!("`{name}`"))
+    } else {
+        name
```

Comment:
Returning `Cow<str>` here makes a lot of sense, however needing to pass one in strikes me as a bit of an anti-pattern. It looks like that might be used to avoid lifetime issues with the `format!` calls above and "value dropped while borrowed" or something? If so, would it be reasonable to take an `Into<Cow>` to avoid needing to update all of the call sites? I'm thinking something like:

```rust
fn swift_keyword_aware_rename<'a, T>(name: T) -> Cow<'a, str>
where
    T: Into<Cow<'a, str>>,
{
    let name = name.into();
    if SWIFT_KEYWORDS.contains(&name.as_ref()) {
        Cow::Owned(format!("`{name}`")
    } else {
        name
    }
}
```

That way we still avoid the extra allocations, but we can directly pass either a `String` or `&str` into the method without needing to explicitly annotate `Cow::Owned` or `Cow::Borrowed` at the call site.

Reply:
Yup I felt I missed something when repeatedly wrapping in `Cow`. Pushed update.